### PR TITLE
Integrated with QubeWire from KeySmith

### PR DESCRIPTION
**Pull request by RajaBoseLeoM**
## What does this PR contains?

1. Name changed from KeySmith to QubeWire.
2. Cleanup few changes in QubeWireClient and made it to work

## Usage example

bash$./QubeWireClient **<Client-Id>**
QubeWire sign-in page opened in web browser. Please sign-in to proceed.
Waiting for user to sign-in...
Successfully signed in as Distributor@gmail.com (World distributor)

1. Sign PKL/CPL.
2. Upload DKDM.
3. Quit.

Please select an action? 1
Enter unsigned CPL/PKL file path? /test/DCP/Unsigned/Movie1.cpl.xml
Uploading CPL/PKL to QubeWire for signing...
Waiting for QubeWire to sign the CPL/PKL...
CPL/PKL successfully signed and available here /test/DCP/Unsigned/Movie1.cpl.signed.xml

1. Sign PKL/CPL.
2. Upload DKDM.
3. Quit.

Please select an action? 3

@sivachandran

### CODE REVIEW

File: README.txt [hunk 1]
```diff
@@ -1,21 +1,21 @@
 Overview
 ========
-KeySmithClient is a C++ application demonstrating KeySmith mastering APIs. It demonstrates the following functionalities of a mastering client.
+QubeWireClient is a C++ application demonstrating QubeWire mastering APIs. It demonstrates the following functionalities of a mastering client.
     - Logging in and acquiring acess token
```
Comment (sivachandran) [inline README.txt]: fix typo `acess`

File: README.txt [hunk 2]
```diff
@@ -1,21 +1,21 @@
 Overview
 ========
-KeySmithClient is a C++ application demonstrating KeySmith mastering APIs. It demonstrates the following functionalities of a mastering client.
+QubeWireClient is a C++ application demonstrating QubeWire mastering APIs. It demonstrates the following functionalities of a mastering client.
```
Comment (sivachandran) [inline README.txt]: `Qube Wire`, two words. Don't change the `QubeWireClient` as it is the program name.

File: src/QubeWireClient.cpp [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+
+const string QUBEWIRE_PRODUCT_ID = "07c0e191-c79c-48c2-8d93-43e2a67ef1d0";
+const string QUBEWIRE_URL = "https://api.qubewire.com";
+const string QUBEACCOUNT_URL = "https://account.qubecinema.com";
+
+struct QubeWireClient::Impl
+{
+    Impl(const string& clientId)
+    {
+        // Make sure you maintain consistensy in QubeStoreClient::Impl() before making any changes
```
Comment (sivachandran) [inline src/QubeWireClient.cpp]: remove this comment

File: src/QubeWireClient.cpp [hunk 4]
```diff
@@ -0,0 +1,509 @@
...
+
+        throw runtime_error(_GetErrorMessage(response));
+    }
+
+private:
+    void _InitializeHttpClient()
+    {
+        http::client::options options;
+
+#ifndef USE_STAGING
```
Comment (sivachandran) [inline src/QubeWireClient.cpp]: remove this macro

File: src/QubeWireClient.cpp [hunk 5]
```diff
@@ -0,0 +1,509 @@
...
+        requestBody << "code=" << _sessionId << "&client_id=" << _clientId
+                    << "&client_secret=null&grant_type=authorization_code&access_type=offline";
+
+        http::client::response response =
+            _PostRequest(_pollingEndpoint, requestBody.str(), "application/x-www-form-urlencoded");
+
+        if (status(response) != http_response::accepted && status(response) != http_response::ok)
+        {
+            string errorMessage = to_lower_copy(_GetErrorMessage(response));
+            if (strstr(errorMessage.c_str(), "user denied access"))
```
Comment (sivachandran) [inline src/QubeWireClient.cpp]: simply return the error message

File: src/QubeWireClient.cpp [hunk 6]
```diff
@@ -0,0 +1,509 @@
...
+        {
+            throw runtime_error(_GetErrorMessage(response));
+        }
+        _tokenType = _ParseJsonProperty(body(response), "token_type");
+
+        return _ParseJsonProperty(body(response), "access_token");
+    }
+
+    string _ParseAuthorizeInfo(const string& json, const string& propertyName)
+        {
```
Comment (sivachandran) [inline src/QubeWireClient.cpp]: fix indentation

File: src/QubeWireClient.h [hunk 7]
```diff
@@ -0,0 +1,124 @@
+/**
+ * @file QubeWireClient.h
+ * @author  Qube Dev <QubeDev@realimage.com>
```
Comment (sivachandran) [inline src/QubeWireClient.h]: remove this line

File: src/QubeWireClient.h [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+     *
+     * @param[out] emailId user's email id
+     * @param[out] companyName Company name of the user certificate
+     */
+    void GetUserInfo(std::string& emailId, std::string& companyName);
+
+    /**
+     * Get certificate chain of active company set by user.
+     * The certificate chain contains certificates of leaf, intermediate and root in PEM
+     * (Privacy-enhanced Electronic Mail) form concatenated together
```
Comment (sivachandran) [inline src/QubeWireClient.h]: remove `(Privacy-enhanced Electronic Mail)`

**Review by sivachandran — CHANGES_REQUESTED** [review]

File: src/QubeWireClient.h (hunk 7 above)
Comment (RajaBoseLeoM) [reply src/QubeWireClient.h]: I saw this line and thought its fine, but after seeing your comment it looks like its not fine to have an email address

File: src/QubeWireClient.cpp (hunk 4 above)
Comment (RajaBoseLeoM) [reply src/QubeWireClient.cpp]: Oops: Searched and removed but missed this one

**Review by sivachandran — APPROVED** [review]

**Pull request merged**